src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_file_request.cc

```diff
@@ -0,0 +1,309 @@
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+  smart_objects::SmartObject response_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "Application is not registered");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  // Check that file name param exists else Return
+  LOG4CXX_DEBUG(logger_, "Check if file name param exists");
+  if (!(*message_)[strings::msg_params].keyExists(strings::file_name)) {
+    LOG4CXX_ERROR(logger_, "File name parameter not specified");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "No file name",
+                 &response_params);
+    return;
+  }
+
+  file_name_ = (*message_)[strings::msg_params][strings::file_name].asString();
+
+  if (!file_system::IsFileNameValid(file_name_)) {
+    LOG4CXX_ERROR(logger_,
+                  "File name " << file_name_ << " contains forbidden symbols.");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "File name contains forbidden symbols",
+                 &response_params);
+    return;
+  }
+
+  // Initialize other params with default values. If exists overwrite the values
+  LOG4CXX_DEBUG(logger_, "Intialize non manadatory params with default values");
+
+  if ((*message_)[strings::msg_params].keyExists(strings::file_type)) {
+    file_type_ = static_cast<mobile_apis::FileType::eType>(
+        (*message_)[strings::msg_params][strings::file_type].asInt());
+  }
+
+  if ((*message_)[strings::msg_params].keyExists(strings::length)) {
+    length_ = (*message_)[strings::msg_params][strings::length].asInt();
+  }
+
+  if ((*message_)[strings::msg_params].keyExists(strings::offset)) {
+    offset_ = (*message_)[strings::msg_params][strings::offset].asInt();
+  }
+
+  // Check if file exists on system (may have to use app service id to get the
+  // correct app folder)
+  LOG4CXX_DEBUG(logger_, "Check if file exists on system");
+  std::string file_path;
+  bool forward_to_hmi;
+
+  if (GetFilePath(file_path, forward_to_hmi)) {
+    if (forward_to_hmi) {
+      LOG4CXX_DEBUG(logger_, "Forwarding GetFile request to HMI");
+      SendHMIRequest(hmi_apis::FunctionID::BasicCommunication_GetFileFromHMI,
+                     &(*message_)[strings::msg_params],
+                     true);
+      return;
+    }
+  } else {
+    LOG4CXX_ERROR(logger_, "Could not get file path");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "Could not get file path",
+                 &response_params);
+  }
+
+  const std::string full_path = file_path + "/" + file_name_;
+  if (!file_system::FileExists(full_path)) {
+    LOG4CXX_ERROR(logger_, "File " << full_path << " does not exist");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
```

Comment:
`FILE_NOT_FOUND` would probably be the best result code in this case.

Reply:
I agree. I'd have to add that result code the response. According to the spec the result codes are ```<param name="resultCode" type="Result" platform="documentation" mandatory="true">
		<description>See Result</description>
		<element name="SUCCESS"/>
		<element name="REJECTED"/>
		<element name="DISALLOWED"/>
		<element name="INVALID_DATA"/>
		<element name="OUT_OF_MEMORY"/>
		<element name="TOO_MANY_PENDING_REQUESTS"/>
		<element name="APPLICATION_NOT_REGISTERED"/>
		<element name="GENERIC_ERROR"/>
	</param>```

Reply:
Hmmm, I don't think this is for anything other than documentation, so I don't think it is a huge deal if we add a result to this list.